src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/impl/HBaseRepositoryImpl.kt

```diff
@@ -70,15 +98,38 @@ class HBaseRepositoryImpl(private val connection: Connection, private val tableN
 
     private var _mutableMap: MutableMap<String, Boolean> = mutableMapOf()
 
-    private fun get(id: String, version: Long) = Get(tableNameUtil.decodePrintable(id)).apply {
+    private fun get(id: String, version: Long, replicaId: Int) = Get(tableNameUtil.decodePrintable(id)).apply {
         setTimeStamp(version)
         isCheckExistenceOnly = true
         consistency = Consistency.TIMELINE
+        if (replicaId >= 0) {
+            setReplicaId(replicaId)
+        }
     }
 
     private fun table(topicName: String) = TableName.valueOf(tableName(topicName))
     private fun tableName(topicName: String) = tableNameUtil.getTableNameFromTopic(topicName)
 
+    fun randomiseReplicaId(replicationFactor: Int): Int {
+        if (replicationFactor == -1) {
+            return -1
+        }
```

Comment:
what's the -1 check for?

Reply:
Checks for the default value defined if no env var is provided for replication.